### PR TITLE
Fixes recursion error in  Limit

### LINKED ISSUE #18442: RuntimeError in limit
`Limit(tan(x)**(2**(sqrt(pi))), x, oo, dir='-')` gives `RuntimeError`

### PR DESCRIPTION
**Pull request by Smit-create**
Fixes recursion error in  Limit
#### References to other Issues or PRs

Fixes #18442 

#### Brief description of what is fixed or changed
Before fixing:
```
>>> limit(tan(x)**(2**(sqrt(pi))), x, oo, dir='-')
RecursionError: maximum recursion depth exceeded
```

After fixing:
```
>>> limit(tan(x)**(2**(sqrt(pi))), x, oo, dir='-')
AccumBounds(-oo, oo)*AccumBounds(-oo, oo)**(2**(sqrt(pi)))
```
#### Release Notes

* calculus
  * Fixes Recursion Error in `AccumulationBounds.__pow__` by adding a check on denominator

### CODE REVIEW

**Comment by jksuom** [comment]
The release notes should say that the recursion error occurs in `AccumBounds.__pow__`. It is only coincidental that it appears when taking a limit.

**Comment by jksuom** [comment]
It would be nice to have a more meaningful limit. How about testing the following patch:
```
--- a/sympy/calculus/util.py
+++ b/sympy/calculus/util.py
@@ -1341,7 +1341,7 @@ def __pow__(self, other):
                 if den!=1:
                     num_pow = self**num
                     return num_pow**(1 / den)
-            return Pow(self, other, evaluate=False)
+            return AccumBounds(-oo, oo)
 
         return NotImplemented
```
Some information will be lost but I don't think that it would be useful anyway. The results of some tests may change but it should be possible to edit them.

**Comment by Smit-create** [comment]
Yes, I will test with the changes

File: sympy/series/tests/test_limits.py [hunk 1]
```diff
@@ -608,3 +608,7 @@ def test_issue_17671():
 
 def test_issue_18306():
     assert limit(sin(sqrt(x))/sqrt(sin(x)), x, 0, '+') == 1
+
+def test_issue_18442():
+    assert limit(tan(x)**(2**(sqrt(pi))), x, oo, dir='-') == \
+            AccumBounds(-oo, oo)*AccumBounds(-oo, oo)**(2**(sqrt(pi)))
```
Comment (jksuom) [inline sympy/series/tests/test_limits.py]: I think that this could be simply `AccumBounds(-oo, oo)`. That is one way of saying that there is no limit.

Comment (Smit-create) [reply sympy/series/tests/test_limits.py]: It is getting simplified to `AccumBounds(-oo, oo)**(1 + 2**(sqrt(pi)))` using
`limit(tan(x)**(2**(sqrt(pi))), x, oo, dir='-').simplify()`

Comment (jksuom) [reply sympy/series/tests/test_limits.py]: In this branch, I get without any simplifications:
```
>>> from sympy import *
>>> x = Symbol('x')
>>> limit(tan(x)**(2**(sqrt(pi))), x, oo, dir='-')
AccumBounds(-oo, oo)
```

Comment (Smit-create) [reply sympy/series/tests/test_limits.py]: I mistakenly checked on the branch without last commit. Thanks :)

**Comment by jksuom** [comment]
Thanks, looks good.

**Pull request merged**